src/make_trade.ts

```diff
@@ -163,3 +176,18 @@ async function signOrder(
   parts.v = parts.v | 0x80;
   return hexlify(concat([parts.r, parts.s, hexValue(parts.v)]));
 }
+
+async function giveAllowanceIfNecessary(
+  sellToken: TokenInfo,
+  sellAmount: BigNumber,
+  trader: SignerWithAddress,
+  allowanceManager: string,
+  ethers: HardhatEthersHelpers
+) {
+  const erc20 = await toERC20(sellToken.address, ethers);
+  const allowance = await erc20.allowance(trader.address, allowanceManager);
```

Comment:
Don't you have to convert the return to BigNumber?

Reply:
If I'm not mistaken, `ethers` will do this by default when ABI decoding the return data.